### ISSUE TITLE
[Share screen] How to control share screen limited?

### ISSUE THREAD

**Issue by mrnav90** [aws/amazon-chime-sdk-js#1595]
Hi Chime SDK Team,

Currently, I know Chime SDK only allows share screen 2 times in every meeting. How can I disable the button that starts the share screen has limited?
I saw `AudioVideoObserver` have event `videoAvailabilityDidChange` this event will receive when meeting room available 25 videos in meeting room so I can disable button `Turn on Camera` when video has limited.
I hope `ContentShareObserver` also have same event like `AudioVideoObserver` then we can disable button share screen.

I also one question: How long until Chime SDK will allow share screen unlimited?

Many thanks.

**Comment by simmkyu** [comment]
> Currently, I know Chime SDK only allows share screen 2 times in every meeting.

Yes, the Chime SDK for JavaScript allows up to two simultaneous screen shares in a single meeting. You can disable or enable screen sharing as many as you want.

⠀

> How can I disable the button that starts the share screen has limited?

You can use the `videoTileDidUpdate` observer method to track how many screen shares are enabled. When you or another attendee enables screen sharing, you will receive the `videoTileDidUpdate` observer method with a video tile object. The video tile object has the `isContent` property.

https://github.com/aws/amazon-chime-sdk-js/blob/710e2bc1d76be115b86d4196363bafc471be9239/src/videotile/VideoTileState.ts#L23-L26

In the [Chime SDK React Components Library](https://github.com/aws/amazon-chime-sdk-component-library-react), for example, uses the `videoTileDidUpdate` method to allow only one screen sharing at a time. [Click here to see the example.](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/89c57ab342b0e7557570a8a231d88dbf6a8bbec7/src/providers/ContentShareProvider/index.tsx#L41-L75)

⠀

> I also one question: How long until Chime SDK will allow share screen unlimited?

We don't have a near-term plan of supporting unlimited content sharing. We'll keep you informed once we have an update.

**Comment by mrnav90** [comment]
Hi @simmkyu 

> You can use the videoTileDidUpdate observer method to track how many screen shares are enabled. When you or another attendee enables screen sharing, you will receive the videoTileDidUpdate observer method with a video tile object. The video tile object has the isContent property.

This is away check normal, it's mean Chime SDK do not support event for share screen limited?
Could you help add this event to feature request?

**Comment by simmkyu** [comment]
The current version of the Chime SDK for JavaScript (2.17.0) does not support the API on [videoAvailabilityDidChange ](https://aws.github.io/amazon-chime-sdk-js/classes/meetingsessionvideoavailability.html) to tell whether the attendee can start screen sharing. I added this issue to the feature request column on our project board and will keep you posted once the API is available.

**Comment by mrnav90** [comment]
@simmkyu 
OK, thank you so much, I hope see it soon >:D<

**Comment by xuesichao** [comment]
Thank you for submitting a feature request. We will track your request via an entry on the [Amazon Chime SDK Project Board](https://github.com/orgs/aws/projects/253) for centralized visibility of the request and its status and close this GitHub issue.
Thank you for your feedback and for contributing to the future of the service.
